packages/api/cms-api/src/auth/services/site-preview.auth-service.ts

```diff
@@ -0,0 +1,45 @@
+import { Injectable, Type } from "@nestjs/common";
+import { Request } from "express";
+import { errors, jwtVerify } from "jose";
+
+import { AuthenticateUserResult, AuthServiceInterface, SKIP_AUTH_SERVICE } from "../util/auth-service.interface";
+
+export interface SitePreviewAuthServiceConfig {
+    sitePreviewSecret: string;
+}
+
+export function createSitePreviewAuthService({ sitePreviewSecret }: SitePreviewAuthServiceConfig): Type<AuthServiceInterface> {
+    @Injectable()
+    class SitePreviewAuthService implements AuthServiceInterface {
+        async authenticateUser(request: Request): Promise<AuthenticateUserResult> {
+            if (typeof sitePreviewSecret !== "string" || sitePreviewSecret.length === 0) {
```

Comment:
Should we add a min length here? Currently " " would be a valid site preview secret.